multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/UnsupportedParameterFinder.java

```diff
@@ -0,0 +1,47 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.resolvers.ReferencesFinder;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+@Named("UnsupportedParameterFinder")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class UnsupportedParameterFinder {
+
+    @Inject
+    private SupportedParameterChecker supportedParameterChecker;
+
+    public List<String> findUnsupportedParameters(DeploymentDescriptor descriptor) {
+        List<String> unknownParametersContainer = findUnsupportedParametersInDescriptor(descriptor);
+        Set<String> references = findReferencesInDescriptor(descriptor);
+        return filterUnsupportedButReferencedParameters(unknownParametersContainer, references);
+    }
+
+    private List<String> findUnsupportedParametersInDescriptor(DeploymentDescriptor descriptor) {
+        List<String> unknownParametersContainer = new ArrayList<>();
+        supportedParameterChecker.fillListWithUnmatched(descriptor, unknownParametersContainer);
```

Comment:
maybe make the fillListWithUnmatched() method just init and return the list internally, same with the .fillWithReferences below. Since we contruct a new instance for the check anyways, no need to pass an empty list to be filled inside

Reply:
maybe also rename the checks' method to be more descriptive - e.g. supportedParameterChecker.getCustomParameters
ReferencesFinder().getAllReferences

Reply:
Done